eZ/Publish/Core/Search/Legacy/Content/Common/Gateway/CriterionHandler/ContentTypeIdentifier.php

```diff
@@ -75,7 +76,15 @@ public function handle(
         $idList = [];
 
         foreach ($criterion->value as $identifier) {
-            $idList[] = $this->contentTypeHandler->loadByIdentifier($identifier)->id;
+            try {
+                $idList[] = $this->contentTypeHandler->loadByIdentifier($identifier)->id;
+            } catch (NotFoundException $e) {
+                // Skip non-existing content types
```

Comment:
Question: wouldn't we actually want to log (warning) usage of non-existing Content Type identifiers? So the Maintainers can trace any outdated apps.

Reply:
@andrerom mentioned it earlier. Yes, but as a separate improvement, that adds logging consistently to every handler that has the same issues.

Reply:
For the case where only some are invalid I agree we should add warning directly here, as it's a bit unique case compared to other criteria.

Maybe something like:
```suggestion
                // Skip non-existing content types, but track for code below
                $invalidIdentifiers[] =  $identifier;
```

Then further down:
```
        if (count($idList) === 0) {
            return $query->expr->not($query->bindValue('1'));
        } else (!empty($invalidIdentifiers)) {
            // log warning or error 
        }
```

Reply:
Updated in https://github.com/ezsystems/ezpublish-kernel/pull/2872/commits/bedbe7f3186b05251de64f78c01d1e82cb0f5712